### PR TITLE
[Feat] [snt9|2_5] Add a case: Inceptionv2-Based Garbage classification identification

### PR DESCRIPTION
**Pull request by ilovexsir**
## Test environment
Hardware: 1*ascend-snt9|ARM: 24核 96GB
Image: python3.9-ms2.5.0-cann8.0.0.beta1

## Test Report

Accuracy: 68.75%
Inference Time: 0.71 ms per image
Training Time: 1h10 minutes
Total Script Runtime: 1h15 minutes
## Related Issues

Fixes #4

### CODE REVIEW

**Comment by hongziqi** [comment]
> ## PR Type
> > Please uncomment only one of the following lines, remove the leading spaces, and leave the other options commented out:
> > /kind ascend-case
> 
> ## Description of Changes
> ## Test environment
> Hardware: 1*ascend-snt9|ARM: 24核 96GB Image: python3.9-ms2.5.0-cann8.0.0.beta1
> 
> ## Test Report
> Accuracy: 10% Inference Time: 0.71 ms per image Training Time: 38 minutes Total Script Runtime: 45 minutes
> 
> ## Related Issues
> Fixes #4
> 
> ## Additional Notes

Consider optimizing the script to improve the model accuracy to at least 50%.

File: dev/ascend/snt9/2_5/Inceptionv2实现垃圾分类识别.ipynb [hunk 1]
```diff
@@ -0,0 +1,806 @@
...
+    "ascend_target = (ms.get_context(\"device_target\") == \"Ascend\")\n",
+    "if ascend_target:\n",
+    "    model = train.Model(network, loss_fn=network_loss, optimizer=network_opt, metrics={\"acc\"}, amp_level=\"O2\")\n",
+    "else:\n",
+    "    model = train.Model(network, loss_fn=network_loss, optimizer=network_opt, metrics={\"acc\"}, amp_level=\"O0\")\n",
+    "\n",
+    "# train model\n",
+    "model.train(epoch_size,\n",
+    "            dataset_train,\n",
+    "            callbacks=[ckpt_callback, LossMonitor(500), TimeMonitor(500)],\n",
```
Comment (hongziqi) [inline dev/ascend/snt9/2_5/Inceptionv2实现垃圾分类识别.ipynb:633]:
```suggestion
    "            callbacks=[ckpt_callback, LossMonitor(step_size), TimeMonitor(step_size)],\n",
```

Comment (ilovexsir) [reply dev/ascend/snt9/2_5/Inceptionv2实现垃圾分类识别.ipynb]: Done

File: dev/ascend/snt9/2_5/Inceptionv2实现垃圾分类识别.ipynb [hunk 2]
```diff
@@ -0,0 +1,877 @@
...
+    "ascend_target = (ms.get_context(\"device_target\") == \"Ascend\")\n",
+    "if ascend_target:\n",
+    "    model = train.Model(network, loss_fn=network_loss, optimizer=network_opt, metrics={\"acc\"}, amp_level=\"O2\")\n",
+    "else:\n",
+    "    model = train.Model(network, loss_fn=network_loss, optimizer=network_opt, metrics={\"acc\"}, amp_level=\"O0\")\n",
+    "\n",
+    "# train model\n",
+    "model.train(epoch_size,\n",
+    "            dataset_train,\n",
+    "            callbacks=[ckpt_callback, LossMonitor(step_size), TimeMonitor(step_size)],\n",
```
Comment (hongziqi) [inline dev/ascend/snt9/2_5/Inceptionv2实现垃圾分类识别.ipynb:704]:
```suggestion
    "            callbacks=[ckpt_callback, LossMonitor(step_size)],\n",
```
Please remove TimeMonitor if epochs > 10.

**Comment by hongziqi** [comment]
It is recommended to supplement the network architecture description.

**Comment by hongziqi** [comment]
> ## PR Type
> > Please uncomment only one of the following lines, remove the leading spaces, and leave the other options commented out:
> > /kind ascend-case
> 
> ## Description of Changes
> ## Test environment
> Hardware: 1*ascend-snt9|ARM: 24核 96GB Image: python3.9-ms2.5.0-cann8.0.0.beta1
> 
> ## Test Report
> Accuracy: 68.75% Inference Time: 0.71 ms per image Training Time: 57 minutes Total Script Runtime: 64 minutes
> 
> ## Related Issues
> Fixes #4
> 
> ## Additional Notes

Please update `Training Time` according to the script to `1h 10min`, and ensure `Total Script Runtime` is updated accordingly.

File: dev/ascend/snt9/2_5/Inceptionv2实现垃圾分类识别.ipynb [hunk 3]
```diff
@@ -0,0 +1,831 @@
...
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "# Inceptionv2 实现垃圾分类识别\n",
+    "该项目可以帮助实现垃圾分类，类别包括：有害垃圾(Hazardous), 厨余垃圾(kitchen),其他垃圾(other),可回收垃圾(Recyclable)。"
```
Comment (hongziqi) [inline dev/ascend/snt9/2_5/Inceptionv2实现垃圾分类识别.ipynb:12]:
```suggestion
    "该项目可以帮助实现垃圾分类，类别包括：有害垃圾(Hazardous)、厨余垃圾(kitchen)、可回收垃圾(Recyclable)和其他垃圾(other)。"
```

Comment (ilovexsir) [reply dev/ascend/snt9/2_5/Inceptionv2实现垃圾分类识别.ipynb]: Done

File: dev/ascend/snt9/2_5/Inceptionv2实现垃圾分类识别.ipynb [hunk 4]
```diff
@@ -0,0 +1,831 @@
...
+    "![conv](https://openi.pcl.ac.cn/attachments/788fa545-fa7c-4c90-9a8e-c1d5c353bcf8?type=0)\n",
+    "\n",
+    "2) 两个 3x3 卷积层作用可以代替一个 5x5 卷积层.\n",
+    "\n",
+    "![](https://openi.pcl.ac.cn/attachments/926f59c3-25a4-401a-b38c-78c9171c7e4a?type=0)\n",
+    "\n",
+    "3) 28x28 的 Inception 模块的数量由 2 增加到了 3.\n",
+    "4) Inception 模块，Ave 和 Max Pooling 层均有用到.\n",
+    "5) 两个 Inception 模块间不再使用 pooling 层；而在模块 3c 和 4e 中的 concatenation 前采用了 stride-2 conv/pooling 层.\n",
+    "6) 网络结构的第一个卷积层采用了深度乘子为 8 的可分离卷积(separable convolution with depth multiplier 8)，减少了计算量，但训练时增加了内存消耗.\n",
```
Comment (hongziqi) [inline dev/ascend/snt9/2_5/Inceptionv2实现垃圾分类识别.ipynb:197]: This description is inconsistent with the actual implementation code, as the code does not apply the depth multiplier.

File: dev/ascend/snt9/2_5/Inceptionv2实现垃圾分类识别.ipynb [hunk 5]
```diff
@@ -0,0 +1,831 @@
...
+    "\n",
+    "    def construct(self, x):\n",
+    "        \"\"\"construct\"\"\"\n",
+    "        x = self.conv_1(x)\n",
+    "        x = self.bn(x)\n",
+    "        x = self.relu(x)\n",
+    "        return x\n",
+    "\n",
+    "\n",
+    "class depthwise_separable_conv(nn.Cell):\n",
```
Comment (hongziqi) [inline dev/ascend/snt9/2_5/Inceptionv2实现垃圾分类识别.ipynb:249]: Inconsistent with the Inception-v2 design: decomposed convolutions better align with the feature extraction logic of the Inception series.

File: dev/ascend/snt9/2_5/Inceptionv2实现垃圾分类识别.ipynb [hunk 6]
```diff
@@ -0,0 +1,831 @@
...
+    "        self.Relu = nn.ReLU()\n",
+    "\n",
+    "    def construct(self, x):\n",
+    "        \"\"\"construct\"\"\"\n",
+    "        if self.is_use_pointwise is not None:\n",
+    "            x = self.conv1(x)\n",
+    "            x = self.conv2(x)\n",
+    "        else:\n",
+    "            x = self.conv1(x)\n",
+    "        x = self.bn(x)\n",
```
Comment (hongziqi) [inline dev/ascend/snt9/2_5/Inceptionv2实现垃圾分类识别.ipynb:275]: The BatchNorm layer is not defined.

File: dev/ascend/snt9/2_5/Inceptionv2实现垃圾分类识别.ipynb [hunk 7]
```diff
@@ -0,0 +1,831 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "# 定义网络\n",
+    "network = inception_v2_base(num_classes=10, dropout_keep_prob=0.8)"
```
Comment (hongziqi) [inline dev/ascend/snt9/2_5/Inceptionv2实现垃圾分类识别.ipynb:470]: You must ensure that `network = inception_v2_base(num_classes=10, dropout_keep_prob=0.8, use_separable_conv=True)` runs correctly.

Comment (ilovexsir) [reply dev/ascend/snt9/2_5/Inceptionv2实现垃圾分类识别.ipynb]: I can run it.

Comment (ilovexsir) [reply dev/ascend/snt9/2_5/Inceptionv2实现垃圾分类识别.ipynb]: I update it.

**Review by hongziqi — APPROVED** [review]
